drivers/spi/spi_silabs_eusart.c

```diff
@@ -23,25 +34,56 @@ LOG_MODULE_REGISTER(spi_silabs_eusart, CONFIG_SPI_LOG_LEVEL);
 
 #include "spi_context.h"
 
+/* Macro Declarations */
+
+
 #define SPI_WORD_SIZE 8
 
-/* Structure Declarations */
+#if	defined(CONFIG_SPI_ASYNC) && defined(SPI_SILABS_SPI_EUSART_DMA)
```

Comment:
One of these conditions seems redundant since you're always enabling SPI_SILABS_SPI_EUSART_DMA together with CONFIG_SPI_ASYNC. Once you've added the new Kconfig option this should just be `#if defined(CONFIG_SPI_SILABS_EUSART_DMA)`